lib/cache/cache.go

```diff
@@ -90,6 +90,20 @@ func ForNode(cfg Config) Config {
 	return cfg
 }
 
+// ForKubernetes sets up watch configuration for a kubernetes service.
+func ForKubernetes(cfg Config) Config {
```

Comment:
Something I stumbled upon while working on db access is that I also had to add `KindDatabaseServer` (my new server type) to `ForAuth()`, `ForProxy()` and `ForRemoteProxy()` in this module, otherwise it would not be updated in cache. I'd check if you need to do the same for your `KindKubeService`.